### PR TITLE
drivers: sensor: adt7420: Add multi-instance support

### PR DESCRIPTION
**Pull request by benjaminbjornsson**
Move driver to use DT_INST_FOREACH_STATUS_OKAY to add
multi-instance support.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
Should probably update `adt7420_trigger_set` to have a check like:

```
if (!config->interrupt.port) {
   return -ENOSUP;
}
```

**Comment by benjaminbjornsson** [comment]
> Should probably update `adt7420_trigger_set` to have a check like:
> 
> ```
> if (!config->interrupt.port) {
>    return -ENOSUP;
> }
> ```

Agree, fixed!

**Review by galak — DISMISSED** [review]

File: drivers/sensor/adt7420/adt7420.c [hunk 1]
```diff
@@ -209,15 +212,19 @@ static int adt7420_init(const struct device *dev)
...
-		    &adt7420_driver_api);
+#define ADT7420_DEFINE(inst)									\
+	static struct adt7420_data adt7420_data_##inst;						\
+												\
+	static const struct adt7420_dev_config adt7420_config_##inst = {			\
+		.i2c = I2C_DT_SPEC_INST_GET(inst),						\
+												\
+	IF_ENABLED(CONFIG_ADT7420_TRIGGER,							\
+		   (COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, int_gpios),				\
+				(.int_gpio = GPIO_DT_SPEC_INST_GET(inst, int_gpios),), ())))	\
```
Comment (gmarull) [inline drivers/sensor/adt7420/adt7420.c:223]: `GPIO_DT_SPEC_INST_GET_OR`

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/sensor/adt7420/adt7420.c (hunk 1 above)
Comment (benjaminbjornsson) [reply drivers/sensor/adt7420/adt7420.c]: Yes, that's way cleaner :) Thanks!

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**